### PR TITLE
top-workflows-with-longest-execution-time

### PR DESCRIPTION
**Pull request by oanaBiancaSchiopu**

### CODE REVIEW

File: rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/dto/FilteredTopWorkflowExecutionTimeData.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ */
+package org.ow2.proactive_grid_cloud_portal.scheduler.dto;
+
+public class FilteredTopWorkflowExecutionTimeData implements java.io.Serializable {
```
Comment (fviale) [inline rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/scheduler/dto/FilteredTopWorkflowExecutionTimeData.java:28]: this name is a bit long, why not simply WorkflowExecutionTimeData

File: scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/FilteredTopWorkflowExecutionTime.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ */
+package org.ow2.proactive.scheduler.common.job;
+
+import java.io.Serializable;
+
+import org.objectweb.proactive.annotation.PublicAPI;
+
+
+@PublicAPI
+public class FilteredTopWorkflowExecutionTime implements Serializable {
```
Comment (fviale) [inline scheduler/scheduler-api/src/main/java/org/ow2/proactive/scheduler/common/job/FilteredTopWorkflowExecutionTime.java:34]: WorkflowExecutionTime ?

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 3]
```diff
@@ -897,46 +899,17 @@ public List<FilteredTopWorkflow> getTopWorkflowsWithIssues(int numberOfWorkflows
...
-                    query.setParameter("tenant", tenant);
-                }
-            }
-            if (startTime != 0) {
-                query.setParameter("startTime", startTime);
-            }
-            if (endTime != 0) {
-                query.setParameter("endTime", endTime);
-            }
+            String selectQuery = "select jobName, projectName, (sum(numberOfFailedTasks) + sum(numberOfFaultyTasks)) as errorCount, count(*) as numberOfExecution from JobData where (numberOfFailedTasks > 0 or numberOfFaultyTasks > 0) ";
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:902]: could you please add indexes (in JobData class) to numberOfFailedTasks and numberOfFaultyTasks columns ?

Comment (fviale) [reply scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:902]: selectQuery -> selectSubQuery

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 4]
```diff
@@ -897,46 +899,17 @@ public List<FilteredTopWorkflow> getTopWorkflowsWithIssues(int numberOfWorkflows
...
+            String groupByQuery = "group by jobName, projectName order by errorCount desc";
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:903]: groupByQuery -> subQueryGroupByStatement

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 5]
```diff
@@ -948,6 +921,78 @@ public List<FilteredTopWorkflow> getTopWorkflowsWithIssues(int numberOfWorkflows
...
+            queryString.append("and owner = :user ");
+            if (hasTenant) {
+                queryString.append("and (tenant = :tenant or tenant = null) ");
+            }
+        }
+        if (startTime != 0) {
+            queryString.append("and finishedTime >= :startTime ");
+        }
+        if (endTime != 0) {
+            queryString.append("and finishedTime <= :endTime ");
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:972]: and finishedTime < :endTime

File: scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java [hunk 6]
```diff
@@ -948,6 +921,78 @@ public List<FilteredTopWorkflow> getTopWorkflowsWithIssues(int numberOfWorkflows
...
+        }
+        queryString.append(groupByQuery);
+
+        Query query = session.createQuery(queryString.toString());
```
Comment (fviale) [inline scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/db/SchedulerDBManager.java:976]: I don't see the notion of select .... from (select ... from JobData) as something

**Review by tao-xinxiu — APPROVED** [review]

**Review by fviale — APPROVED** [review]

**Pull request merged**